src/components/vouchers/VoucherListItem.tsx

```diff
@@ -0,0 +1,114 @@
+import { StackScreenProps } from '@react-navigation/stack';
+import React, { memo } from 'react';
+import { Platform, StyleSheet, View } from 'react-native';
+import { Card } from 'react-native-elements';
+
+import { Icon, colors, normalize } from '../../config';
+import { imageHeight, imageWidth } from '../../helpers';
+import { QUERY_TYPES } from '../../queries';
+import { TVoucherItem } from '../../types';
+import { Image } from '../Image';
+import { BoldText, RegularText } from '../Text';
+import { Touchable } from '../Touchable';
+
+import { Discount } from './Discount';
+
+export const VoucherListItem = memo(
+  ({
+    navigation,
+    horizontal,
+    item
+  }: {
+    navigation: StackScreenProps<any>;
+    horizontal: boolean;
+    item: TVoucherItem;
+  }) => {
+    const {
+      discountType,
+      id,
+      // TODO: update this code when the image is added on the server
+      image = { uri: 'https://picsum.photos/500/300' },
+      params,
+      quota,
+      routeName: name,
+      subtitle,
+      title
+    } = item;
+
+    return (
+      <Touchable onPress={() => navigation && navigation.push(name, params)} disabled={!navigation}>
+        <Card containerStyle={styles.container}>
+          {!!image?.uri && (
+            <Image
+              source={{ uri: image.uri }}
+              style={stylesWithProps({ horizontal }).image}
+              containerStyle={styles.imageContainer}
+              borderRadius={5}
+            />
+          )}
+
+          {!!discountType && (
+            <Discount discount={discountType} query={QUERY_TYPES.VOUCHERS} id={id} />
+          )}
+
+          <View
+            style={{ flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center' }}
+          >
```

Comment:
this `View` should be possible with our `WrapperRow spaceBetween style={{ alignItems: 'center' }}` to reuse more of our components, right?

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/988/commits/de161547c3488ae2a87cbe6a5163c6cea10564e2